### ISSUE TITLE
[BUG] Possible bug in flash_attn_func sliding window attention when window_size is enabled (unexpected zero output)

### ISSUE THREAD

**Issue by Ghjk94522** [Dao-AILab/flash-attention#2199]
I encountered a potentially unexpected behavior when testing the window_size feature of flash_attn_func, and I am not sure whether this indicates a bug or a misunderstanding on my side regarding the window definition.

## Reproduction
I am testing flash_attn_func with window_size enabled (i.e., not (-1, -1)) to use sliding window local attention. The minimal test code is as follows:
``` python
from flash_attn import flash_attn_func
import torch

def test(batch_size, seqlen_q, seqlen_k, nhq, nhk, d_q, d_v, device, dtype):
    q = torch.randn(
        batch_size, seqlen_q, nhq, d_q, device=device, dtype=dtype
    )
    k = torch.randn(
        batch_size, seqlen_k, nhk, d_q, device=device, dtype=dtype
    )
    v = torch.randn(
        batch_size, seqlen_k, nhk, d_v, device=device, dtype=dtype
    )
    
    softmax_scale = q.shape[-1] ** (-0.5)
    window_size = [32, 64]
    
    out_fa, lse, _  = flash_attn_func(
        q,
        k,
        v,
        causal=False,
        softmax_scale=softmax_scale,
        window_size=window_size,
        dropout_p=0.0,
        return_attn_probs=True,
    )
    
    print(f'{lse=}')
    print(f'{out_fa=}')
    print(f'{out_fa[0][2][0]=}')


if __name__ == '__main__':
    test(
        batch_size=1,
        seqlen_q=129,
        seqlen_k=63,
        nhq=1,
        nhk=1,
        d_q=128,
        d_v=128,
        device='cuda',
        dtype=torch.float16,
    )
```
In this test, the output out_fa[0][2] is entirely zero.

## Expected behavior
According to the [documentation](https://github.com/Dao-AILab/flash-attention/blob/main/flash_attn/flash_attn_interface.py#L1100) of window_size:

> If window_size != (-1, -1), implements sliding window local attention. Query at position i will only attend to keys between `[i + seqlen_k - seqlen_q - window_size[0], i + seqlen_k - seqlen_q + window_size[1]]` inclusive.

In my case:
- seqlen_q = 129
- seqlen_k = 63
- window_size = (32, 64)
- i_q = 2

The computed key range should therefore be:
```
[2 + 63 - 129 - 32, 2 + 63 - 129 + 64] = [-96, 0]
```

After clipping to valid indices, this range includes k[0], so at least one key should participate in the attention computation. Based on this, I would not expect the output at out_fa[0][2] to be entirely zero.

## Actual behavior
However, the actual output of my test is:
```
lse=tensor([[[   inf,    inf,    inf, 0.6316, 1.6246, 0.4839, 1.3593, 0.9513,
          2.6343, 2.3098, 2.7766, 2.4007, 2.8883, 2.7665, 3.1259, 3.3332,
          3.4171, 3.4124, 2.9602, 3.2019, 3.2477, 3.4286, 3.5214, 3.9459,
          3.9400, 3.6225, 3.6336, 3.8745, 3.4689, 3.5932, 3.5084, 3.9032,
          3.9372, 3.5755, 3.9813, 3.6754, 4.1653, 4.3695, 4.0190, 4.2652,
          4.2419, 4.1991, 4.0894, 4.5906, 4.3435, 4.2170, 4.0975, 4.0660,
          4.0357, 4.3031, 4.3524, 4.5284, 4.2557, 4.7080, 4.3081, 4.3329,
          4.3200, 4.6012, 4.4780, 4.8695, 4.4995, 4.6672, 4.6180, 4.8081,
          4.5913, 4.4958, 4.5158, 4.7530, 4.5411, 4.7149, 4.5636, 4.5031,
          4.4412, 4.5339, 4.7504, 4.3508, 4.6666, 4.5823, 4.6396, 4.7633,
          4.8401, 4.5794, 4.5284, 4.8403, 4.6640, 4.7735, 4.6316, 4.8825,
          4.6029, 4.7829, 4.5338, 4.8080, 4.5666, 4.5575, 4.6640, 4.8132,
          4.3923, 4.6827, 4.4160, 4.8774, 4.7384, 4.3558, 4.6228, 4.4751,
          4.6023, 4.6263, 4.6346, 4.4839, 4.3007, 4.4169, 4.3605, 4.3673,
          4.5481, 4.2500, 3.9272, 4.4302, 4.4592, 4.3996, 4.2062, 3.9491,
          3.8932, 4.0842, 4.0248, 3.9096, 3.9983, 4.1945, 3.7596, 4.3359,
          4.1391]]], device='cuda:0')
out_fa=tensor([[[[ 0.0000,  0.0000,  0.0000,  ...,  0.0000,  0.0000,  0.0000]],

         [[ 0.0000,  0.0000,  0.0000,  ...,  0.0000,  0.0000,  0.0000]],

         [[ 0.0000,  0.0000,  0.0000,  ...,  0.0000,  0.0000,  0.0000]],

         ...,

         [[-0.0591, -0.3464,  0.4092,  ..., -0.0892, -0.3877,  0.5391]],

         [[ 0.1204, -0.0538,  0.0053,  ..., -0.0021, -0.4500,  0.6797]],

         [[-0.1411,  0.0386,  0.1071,  ...,  0.0206, -0.3726,  0.2905]]]],
       device='cuda:0', dtype=torch.float16)
out_fa[0][2][0]=tensor([0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0., 0.,
        0., 0., 0., 0., 0., 0., 0., 0.], device='cuda:0', dtype=torch.float16)

```
which shows that out_fa[0][2] is all zeros.

## Environment
- flash_attn: v2.8.3
- torch: v2.8.0
- GPU: Nvidia A10
- Nvidia Driver Version: 580.65.06      
- CUDA Version: 13.0

## Question
Could someone clarify whether this behavior is expected?
Specifically:
- Is there an additional constraint or masking behavior in flash_attn_func when window_size is used that would fully mask this query position?
- Or does this indicate a potential bug in the sliding window attention implementation?

Any clarification or guidance would be greatly appreciated.

**Comment by Ghjk94522** [comment]
Or, maybe window_size should be <= seqlen_q and seqlen_k?
